csv/shared/src/main/scala/fs2/data/csv/RowF.scala

```diff
@@ -139,7 +140,13 @@ case class RowF[H[+a] <: Option[a], Header](values: NonEmptyList[String],
     */
   def dropHeaders: Row = Row(values)
 
-  private def byHeader(implicit hasHeaders: HasHeaders[H, Header]): Map[Header, String] =
+  // let's cache this to avoid recomputing it for every call to `as` or similar method
+  // the `Option.get` call is safe since this field is only called in a context where a `HasHeaders`
+  // instance is provided, meaning that the `Option` is `Some`
+  // of course using a lazy val prevents us to make this constraint statistically checked, but
+  // the gain is significant enough to allow for this local unsafety
+  @deprecated("Have you checked that you have a `HasHeaders` instance in scope?")
+  private lazy val byHeader: Map[Header, String] =
```

Comment:
The change is definitely fine (thank you), but maybe we can protect it by indirection via a def? So we would keep `@inline def byHeader` to check the implicit evidence, but it's implementation would be backed by a `private lazy val dontUseMeByHeader`? In the end, it doesn't change anything if we always keep the constraint in mind when changing the class' internals, but I prefer safe over sorry…

Reply:
This is my first implementation. And then the compilation overhead seemed not worth it to me. Even thought about having a private object holding this private field and having one public method. In the end, it's all the same, and I think the warning makes it _safe enough_ ™️

Reply:
OK, fair enough. LGTM then!